pkg/olm/clusterserviceversionlist.go

```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
 
 	return csvObjects, nil
 }
+
+// ListClusterServiceVersionWithNamePattern returns a cluster-wide csv inventory filtered by the name pattern.
+func ListClusterServiceVersionWithNamePattern(
+	apiClient *clients.Settings,
+	namePattern, nsname string,
+	options metaV1.ListOptions) ([]*ClusterServiceVersionBuilder, error) {
+	if namePattern == "" {
+		glog.V(100).Info("The namePattern field to filter-out all relevant csv could not be empty")
+
+		return nil, fmt.Errorf("the namePattern field to filter-out all relevant csv could not be empty")
+	}
+
+	glog.V(100).Infof("Listing csv filtered by the name pattern %s are found in %s namespace", namePattern, nsname)
+
+	notFilteredCsvList, err := ListClusterServiceVersion(apiClient, nsname, options)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list all clusterserviceversions in %s namespace due to %s",
```

Comment:
I think you should standardize on either csvs or clusterserviceversions. Try not to mix them for the sake of consistency.